components/ratgdo/ratgdo.cpp

```diff
@@ -675,39 +676,36 @@ namespace ratgdo {
         }
 
         auto delta = position - this->doorPosition;
+        if (delta == 0) {
+            ESP_LOGD(TAG, "Door is already at position %.2f", position);
+            return;
+        }
+
+        auto duration = delta > 0 ? this->openingDuration : this->closingDuration;
+        if (duration == 0) {
+            ESP_LOGW(TAG, "I don't know duration, ignoring move to position");
+            return;
+        }
+        
         if (delta > 0) { // open
-            if (this->openingDuration==0) {
-                ESP_LOGW(TAG, "I don't know opening duration, ignoring move to position");
-                return;
-            }
-            auto opening_time = this->openingDuration * 1000 * delta;
             doorCommand(data::DOOR_OPEN);
-            this->movingToPosition = true;
-            set_timeout("move_to_position", opening_time, [=] {
-                doorCommand(data::DOOR_STOP);
-                this->movingToPosition = false;
-                this->doorPosition = position;
-            });
             this->positionSyncWhileOpening(delta);
-        } else if (delta < 0) { // close
-            if (this->closingDuration==0) {
-                ESP_LOGW(TAG, "I don't know closing duration, ignoring move to position");
-                return;
-            }
-            delta = -delta;
-            auto closing_time = this->closingDuration * 1000 * delta;
+        } else { // close
             doorCommand(data::DOOR_CLOSE);
-            this->movingToPosition = true;
-            set_timeout("move_to_position", closing_time, [=] {
-                doorCommand(data::DOOR_STOP);
-                this->movingToPosition = false;
-                this->doorPosition = position;
-            });
-            this->positionSyncWhileClosing(delta);
+            this->positionSyncWhileClosing(-delta);
         }
+
+        auto operation_time = duration * 1000 * delta;
```

Comment:
I think on close `set_timeout` could have been called with a negative value before?

```
            delta = -delta;
            auto closing_time = this->closingDuration * 1000 * delta;
```

Reply:
Before it was in the `if (delta<0)` condition body, but in this PR seems like you're not accounting got that.